app/views/RoomMembersView/index.js

```diff
@@ -134,16 +138,71 @@ class RoomMembersView extends React.Component {
 		}
 	}
 
+	async setPermissions() {
+		try {
+			const { navigation } = this.props;
+			const { rid } = navigation.state.params;
+			const permissions = ['set-owner', 'set-leader', 'set-moderator', 'remove-user', 'mute-user'];
+			const result = await RocketChat.hasPermission(permissions, rid);
+			this.setOwnerPermission = result[permissions[0]];
+			this.setLeaderPermission = result[permissions[1]];
+			this.setModeratorPermission = result[permissions[2]];
+			this.removeUserPermission = result[permissions[3]];
+			this.muteUserPermission = result[permissions[4]];
+		} catch (e) {
+			log(e);
+		}
+		Promise.resolve();
+	}
+
 	onLongPressUser = (user) => {
-		if (!this.permissions['mute-user']) {
-			return;
+		if (!this.muteUserPermission) {
+			return false;
 		}
 		const { room } = this.state;
-		const { muted } = room;
-
+		this.isLeader = room && room.roles && room.roles.length && !!room.roles.find(role => role === 'leader');
+		this.isOwner = room && room.roles && room.roles.length && !!room.roles.find(role => role === 'owner');
+		this.isModerator = room && room.roles && room.roles.length && !!room.roles.find(role => role === 'moderator');
```

Comment:
`leader`, `owner` and `moderator` could be constants, as you're going to use them below.